src/cli/commands/lint.ts

```diff
@@ -109,28 +107,50 @@ linting ./openapi.yaml
   }
 }
 
+async function tryReadOrLog(command: Lint, reader: Function) {
+  try {
+    return await reader();
+  } catch (ex) {
+    if (ex.messages) {
+      command.log(ex.messages[0]);
+      command.error(ex.messages[1]);
+    } else {
+      command.error(ex);
+    }
+  }
+}
+
 async function lint(name: string, flags: any, command: Lint, rules?: RuleCollection) {
   if (flags.verbose) {
     command.log(`Linting ${name}`);
   }
   const spec: IParserResult = await readParsable(name, flags.encoding);
   const spectral = new Spectral();
-  if (rules !== undefined) {
+  command.log('Setting up functions...');
+  if (parseInt(spec.data.swagger) === 2) {
+    command.log('Adding OpenAPI 2.0 (Swagger) functions');
+    spectral.addFunctions(oas2Functions());
+  } else if (parseInt(spec.data.openapi) === 3) {
+    command.log('Adding OpenAPI 3.x functions');
+    spectral.addFunctions(oas3Functions());
+  }
+
+  command.log('Setting up rules...');
+  if (rules) {
     if (flags.verbose) {
       command.log(`Found ${Object.keys(rules).length} rules`);
     }
+    command.log('Applying rules. Automatic rule detection is off.');
   } else {
     if (flags.verbose) {
       command.log('No rules loaded, attempting to detect document type');
     }
     if (parseInt(spec.data.swagger) === 2) {
       command.log('OpenAPI 2.0 (Swagger) detected');
-      spectral.addFunctions(oas2Functions());
-      rules = oas2Rules();
+      rules = await tryReadOrLog(command, async () => await oas2Rules());
```

Comment:
Do we want to let spectral continue if the rules cannot be loaded? 

Should it error if it has 1 of 2 rulesets loaded? 
Should it error if it has 0 of 2 rulesets loaded?

I'd say yes for both. Not just a command.error but a process.exit too.

Reply:
Ahh looks like this is the same behavior as before so this can be handled another time!